### PR TITLE
[RNMobile][Embed block] Fix URL update when edited after setting a bad URL of a provider

### LINKED ISSUE #34792: Embed block: URL is not updated when edited after setting a bad URL of a provider
## Description

The embed URL is not modified when edited if a bad URL of a provider was previously set.

## Step-by-step reproduction instructions

1. Add an Embed block.
2. Set a bad URL of any provider (example: https://youtu.be/BAD_URL).
3. Observe that no error message is thrown because we're getting a valid preview response although the content is not.
4. Tap on the pencil icon located in the toolbar to edit the URL.
5. Modify the URL and apply the change by dismissing the bottom sheet.
6. Tap on the pencil icon again.
7. Observe that the URL hasn't been changed and remains with the original value.

## Expected behaviour

The embed URL should be modified when edited.

## Actual behaviour

The embed URL is not modified when edited.

## WordPress information
- WordPress version: N/A
- Gutenberg version: N/A
- Are all plugins except Gutenberg deactivated? N/A
- Are you using a default theme (e.g. Twenty Twenty-One)? N/A

## Device information
- Device: iPhone 12 Pro Max (simulator)
- Operating system: iOS 14.5
- WordPress app version: `develop`

### PR DESCRIPTION
**Pull request by fluiddot**
`gutenberg-mobile` PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/4002

## Description

One of the uses of the `isEditingURL` value is to control the visibility of the embed bottom sheet, as it matches the expected behavior for this. However, I noticed that this value is also used to trigger a side-effect that replaces the block, producing that a new block is added before the URL was actually edited (more info can be found in [this issue](https://github.com/WordPress/gutenberg/issues/34792#issuecomment-924096891)).

This PR introduces a new value in the local state of the embed block to specifically control the visibility of the bottom sheet and prevent undesired side effects.

## How has this been tested?

1. Add an Embed block.
1. Set a bad URL of any provider (example: https://youtu.be/BAD_URL).
1. Observe that no error message is thrown because we're getting a valid preview response although the content is not.
1. Tap on the pencil icon located in the toolbar to edit the URL.
1. Modify the URL and apply the change by dismissing the bottom sheet.
1. Tap on the pencil icon again.
1. Observe that the URL changed to the new value.

## Screenshots 
N/A

## Types of changes

Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/block-library/src/embed/edit.native.js [hunk 1]
```diff
@@ -190,6 +193,10 @@ const EmbedEdit = ( props ) => {
 		}
 	}, [ preview, isEditingURL ] );
 
+	useEffect( () => setShowEmbedBottomSheet( isEditingURL ), [
+		isEditingURL,
+	] );
```
Comment (fluiddot) [inline packages/block-library/src/embed/edit.native.js:200]: This side-effect will assure that the embed bottom sheet is shown/hidden when `isEditingURL` is updated.

File: packages/block-library/src/embed/edit.native.js [hunk 2]
```diff
@@ -282,11 +289,13 @@ const EmbedEdit = ( props ) => {
...
-				isVisible={ isEditingURL }
-				onClose={ () => setIsEditingURL( false ) }
+				isVisible={ showEmbedBottomSheet }
+				onClose={ () => {
+					setShowEmbedBottomSheet( false );
+				} }
 				onSubmit={ ( value ) => {
-					setIsEditingURL( false );
 					setAttributes( { url: value } );
+					setIsEditingURL( false );
```
Comment (fluiddot) [inline packages/block-library/src/embed/edit.native.js]: I found out that the order here is important, we need to update the URL attribute before the following side-effect is triggered when the `isEditingURL` value is changed:
https://github.com/WordPress/gutenberg/blob/6dd74384f726641d2bd40b5d58a3c9f477a1daa0/packages/block-library/src/embed/edit.native.js#L171-L194

File: packages/block-library/src/embed/edit.native.js [hunk 3]
```diff
@@ -282,11 +289,13 @@ const EmbedEdit = ( props ) => {
 			) }
 			<EmbedBottomSheet
 				value={ url }
-				isVisible={ isEditingURL }
-				onClose={ () => setIsEditingURL( false ) }
+				isVisible={ showEmbedBottomSheet }
+				onClose={ () => {
+					setShowEmbedBottomSheet( false );
+				} }
```
Comment (fluiddot) [inline packages/block-library/src/embed/edit.native.js]: The `onClose` prop should only be used for controlling the bottom sheet's visibility.

File: packages/block-library/src/embed/edit.native.js (hunk 2 above)
Comment (ceyhun) [reply packages/block-library/src/embed/edit.native.js]: Do you think maybe we should add a comment here as well?

**Review by ceyhun — APPROVED** [review]
Confirmed the fix both on WPiOS and WPAndroid 👍 LGTM 🎉

File: packages/block-library/src/embed/edit.native.js (hunk 2 above)
Comment (fluiddot) [reply packages/block-library/src/embed/edit.native.js]: Good point, I'll add a comment here to clarify it 👍 .

Comment (fluiddot) [reply packages/block-library/src/embed/edit.native.js]: I added your suggestion in [this commit](https://github.com/WordPress/gutenberg/pull/35013/commits/9bbcc377c711d32f49dcb6b3a7d44fa789274c8a).

**Comment by fluiddot** [comment]
Looks like the `cb() never called!` is striking again, I'm going to apply the workaround for fixing this in the next commits 🔧 .

**Pull request merged**